src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -682,73 +690,107 @@ async function fetchFeatures(params: { extensionPath: string; cwd: string; outpu
+				params.output.write('Appending unauthenticated URIs for v2 and then v1', LogLevel.Trace);
+				tarballUris.push(`${featureSet.sourceInformation.unauthenticatedUri}/${feature.id}.tgz`);
+				tarballUris.push(`${featureSet.sourceInformation.unauthenticatedUri}/${V1_ASSET_NAME}`);
+
 			} else {
-				tarballUri = featureSet.sourceInformation.tarballUri;
+				// We have a plain ol' tarball URI, since we aren't in the github-repo case.
+				tarballUris.push(featureSet.sourceInformation.tarballUri);
 			}
-			
-			if(tarballUri) {
-				const options = {
-					type: 'GET',
-					url: tarballUri,
-					headers
-				};
-				params.output.write(`Fetching tarball at ${options.url}`);
-				params.output.write(`Headers: ${JSON.stringify(options)}`, LogLevel.Trace);
-				const tarball = await request(options, params.output);
-
-				if (!tarball || tarball.length === 0) {
-					params.output.write(`Did not receive a response from tarball download URI`, LogLevel.Error);
-					// Continue loop to the next remote feature.
-					// TODO: Should be more fatal.
-					await cleanupIterationFetchAndMerge(tempTarballPath, params.output);
-					continue;
-				}
 
-				// Filter what gets emitted from the tar.extract().
-				const filter = (file: string, _: tar.FileStat) => {
-					// Don't include .dotfiles or the archive itself.
-					if (file.startsWith('./.') || file === `./${ASSET_NAME}` || file === './.') {
-						return false;
-					}
-					return true;
-				};
+			// Attempt to fetch from 'tarballUris' in order, until one succeeds.
+			for (const tarballUri of tarballUris) {
+				const didSucceed = await fetchContentsAtTarballUri(tarballUri, featCachePath, headers, dstFolder, params.output);
 
-				params.output.write(`Preparing to unarchive received tgz.`, LogLevel.Trace);
-				// Create the directory to cache this feature-set in.
-				await mkdirpLocal(featCachePath);
-				await writeLocalFile(tempTarballPath, tarball);
-				await tar.x(
-					{
-						file: tempTarballPath,
-						cwd: featCachePath,
-						filter
-					}
-				);
-
-				await parseDevContainerFeature(featureSet, feature, featCachePath);
+				if (didSucceed) {
+					params.output.write(`Succeeded fetching ${tarballUri}`, LogLevel.Trace)
+					await parseDevContainerFeature(featureSet, feature, featCachePath);
+					break;
+				}
 			}
-			continue;
+
+			const msg = `(!) Failed to fetch tarball after attempting ${tarballUris.length} possibilities.`;
+			params.output.write(msg, LogLevel.Error);
+			throw new Error(msg);
 		}
 		catch (e) {
-			params.output.write(`Exception: ${e?.Message} `, LogLevel.Trace);
+			params.output.write(`Failed to fetch feature. ${e?.Message ?? ''} `, LogLevel.Trace);
+			// TODO: Should this be more fatal?
 		}
 	}
 }
 
+async function fetchContentsAtTarballUri(tarballUri: string, featCachePath: string, headers: { 'user-agent': string; 'Authorization'?: string; 'Accept'?: string }, dstFolder: string, output: Log): Promise<boolean> {
```

Comment:
reorganized what was previously a part of the above function into its own, so it can be called >1